test/e2etest/kubetest/kubetest_runner.go

```diff
@@ -0,0 +1,231 @@
+		log.Fatal(errors.Wrapf(err, "Couldn't set environment variable KUBERNETES_CONFORMANCE_TEST"))
+	}
+
+	if err = os.Setenv("GINKGO_NO_COLOR", "y"); err != nil {
+		log.Fatal(errors.Wrapf(err, "Couldn't set environment variable GINKGO_NO_COLOR"))
+	}
+}
+
+// DryRun runs kubetest with dryrun=true argument
+func DryRun() (logDir string) {
+	kubetestArgs := createKubetestArgs("", false, true, 1)
+	runKubetest(kubetestArgs)
+	return kubetestArgs.LogDir
+}
+
+// Run runs kubetest to execute e2e testcases for a given testcase description file
+func Run(descFile string) (resultsPath string) {
+	if descFile == "" {
+		log.Fatal("No valid description file provided.")
+	}
+	log.Infof("Running kubetest for %d e2e tests:\n%s", getLinesCount(descFile), getFileContent(descFile))
+
+	parallelTestsFocus, serialTestsFocus := escapeAndConcat(descFile)
+	if parallelTestsFocus != "" {
+		kubtestArgs := createKubetestArgs(parallelTestsFocus, true, false, config.FlakeAttempts)
+		log.Info("Run kubetest in parallel way")
+		log.Infof("kubetest dump dir: %s", kubtestArgs.LogDir)
+		runKubetest(kubtestArgs)
+	}
+	if serialTestsFocus != "" {
+		kubtestArgs := createKubetestArgs(serialTestsFocus, false, false, config.FlakeAttempts)
+		log.Info("Run kubetest in serial way")
+		log.Infof("kubetest dump dir: %s", kubtestArgs.LogDir)
+		runKubetest(kubtestArgs)
+	}
+	return config.LogDir
+}
+
+func getFileContent(filepath string) string {
+	if file, err := os.Open(filepath); err != nil {
+		log.Fatal(err)
+	} else {
+		b, err := ioutil.ReadAll(file)
+		if err != nil {
+			log.Fatal(err)
+		}
+		return string(b)
+	}
+	return ""
+}
+
+func createKubetestArgs(ginkgoFocus string, parallel, dryRun bool, flakeAttempts int) KubetestArgs {
+	ginkgoParallelArg := ""
+	if parallel {
+		ginkgoParallelArg = "--ginkgo-parallel"
+	}
+	if ginkgoFocus != "" {
+		ginkgoFocus = fmt.Sprintf("--ginkgo.focus=%s", ginkgoFocus)
+	}
+	logDir := filepath.Join(config.LogDir, strconv.FormatInt(time.Now().Unix(), 10))
+	_ = os.MkdirAll(logDir, os.FileMode(0777))
+	kubetestArgs := KubetestArgs{ShootConfigPath: config.ShootKubeconfigPath, GinkgoFocus: ginkgoFocus, DryRun: dryRun, LogDir: logDir, FlakeAttempts: flakeAttempts, Provider: Cloudprovider(config.CloudProvider), GinkgoParallel: ginkgoParallelArg}
+	return kubetestArgs
+}
+
+func runKubetest(args KubetestArgs) {
+	ginkgoArgs := fmt.Sprintf("--test_args=--ginkgo.flakeAttempts=%o --ginkgo.dryRun=%t %s", args.FlakeAttempts, args.DryRun, args.GinkgoFocus)
+	cmd := exec.Command("go", "run", "hack/e2e.go", "--", "--provider=skeleton", "--deployment=local", "--test", "--check-version-skew=false", args.GinkgoParallel, ginkgoArgs, fmt.Sprintf("--dump=%s", args.LogDir))
+	cmd.Dir = config.KubernetesPath
+
+	var out bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &out
+	if log.GetLevel() == log.DebugLevel {
+		cmd.Stderr = &stderr
+	}
+	err := cmd.Run()
+
+	if log.GetLevel() == log.DebugLevel {
```

Comment:
just `log.Debug(err.String())`, the check if debug level is already done by the logger itlsef